### PR TITLE
split out server/build/cli modules

### PR DESCRIPTION
**Pull request by ZauberNerd**
Taken from: https://github.com/xing/hops/pull/225

### CODE REVIEW

File: packages/template-react/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   "scripts": {
     "postinstall": "flow-typed install --overwrite",
     "start": "hops start",
+    "build": "hops build",
```
Comment (dmbch) [inline packages/template-react/package.json]: We should switch to production env for builds, shouldn't we?

File: packages/config/readme.md [hunk 2]
```diff
@@ -3,40 +3,43 @@
...
 | `moduleDirs` | `Array<string>` | One or more directories that Hops should use for module path resolution. For example, you can specify a directory that holds a yet unpublished Node.js package. It works similar to the [resolve.modulesDirectories](http://webpack.github.io/docs/configuration.html#resolve-modulesdirectories) field in Webpack, i.e. if you specify `moduleDirs: ["my-module-dir"]`, it will look in `./my-module-dir`, `../my-module-dir`, `../../my-module-dir` and so on |
 | `appDir` | `String` | Directory of your app. This way, you can put your app to a place other than project root. Defaults to `"."` |
 | `buildDir` | `String` | Directory where the build artifacts (i.e. your application/asset files) will be saved to. Defaults to `"build"`. Hops always removes the specified directory before starting a new build |
 | `cacheDir` | `String` | Directory where internally used artifacts not meant to be served will be output (e.g. the `manifest.json`/`manifest.js` files generated by Webpack). Defaults to `"node_modules/.cache/hops"` |
-| `buildConfig` | `String` | Path to your Webpack build configuration file. Defaults to `./node_modules/hops-config/configs/build.js` |
```
[the remark this answers is not in the document]
Comment (ZauberNerd) [reply packages/config/readme.md]: Not sure if I understand correctly - but these three configs `{build,develop,node}Config` have been moved from `hops-config` to `hops-build-config`.

[the remark this answers is not in the document]
Comment (KingHenne) [reply packages/config/readme.md]: I know, but don't I still configure their custom paths using `hops-config` (in the config section of my `package.json`)? How else did hops find them in my project?

[the remark this answers is not in the document]
Comment (ZauberNerd) [reply packages/config/readme.md]: @KingHenne Does this help you?
 https://github.com/xing/hops/blob/e827007f90f5c9a3001dcf672b5ba477440ceaad/packages/build-config/index.js#L6

[the remark this answers is not in the document]
Comment (KingHenne) [reply packages/config/readme.md]: That's what I'm saying, the option is still there and is used. So it shouldn't be removed from the README, right?

[the remark this answers is not in the document]
Comment (ZauberNerd) [reply packages/config/readme.md]: Ah, sorry. I was a bit slow.. :D Yes, I think you are right.

**Comment by KingHenne** [comment]
@dmbch Yes, this should maybe be part of the `hops-build-config` README instead. I had the same thought when I added `apiHost` (basically owned by malt-cli) to the typings of `hops-config`.

**Pull request merged**